models/views/scratch/snowplow_unified_views_this_run.sql

```diff
@@ -115,24 +115,58 @@ with prep as (
 
     , coalesce(t.end_tstamp, p.derived_tstamp) as end_tstamp -- only page views with pings will have a row in table t
 
-    {% if var('snowplow__enable_web') %}
-      , coalesce(t.engaged_time_in_s, 0) as engaged_time_in_s -- where there are no pings, engaged time is 0.
-      , {{ datediff('p.derived_tstamp', 'coalesce(t.end_tstamp, p.derived_tstamp)', 'second') }} as absolute_time_in_s
-      , sd.hmax as horizontal_pixels_scrolled
-      , sd.vmax as vertical_pixels_scrolled
-      , sd.relative_hmax as horizontal_percentage_scrolled
-      , sd.relative_vmax as vertical_percentage_scrolled
+    {% if var('snowplow__enable_web') or var('snowplow__enable_screen_summary_context', false) %}
+      , coalesce(
+        {% if var('snowplow__enable_screen_summary_context', false) %}ss.foreground_sec,{% endif %}
+        t.engaged_time_in_s,
+        0 -- where there are no pings, engaged time is 0.
```

Comment:
Is there a way we could move the screen view logic directly into the engaged time and scroll depth models instead? This feels a little messy to do it here and it would be better for any custom models that they were calculated together in an upstream table?

Reply:
Good call, have included the screen summary data in the engaged time and scroll depth models, I think it looks cleaner now.